tests/core/Logger.test.js

```diff
@@ -0,0 +1,39 @@
+global.chrome = require("sinon-chrome");
+const HFX = require("../../src/HFX");
+
+describe("Logger", () => {
+  it("ToggleDebug works", () => {
+    const startingDebugMode = HFX.Logger.debugMode;
+    expect(HFX.Logger.debugMode).toBe(startingDebugMode);
+    HFX.Logger.toggleDebug();
+    expect(HFX.Logger.debugMode).toBe(!startingDebugMode);
+  });
+
+  it("Error calls console.error", () => {
+    const spy = jest.spyOn(console, "error").mockImplementation();
+    HFX.Logger.error("Error");
+    expect(console.error).toHaveBeenCalledTimes(1);
+    spy.mockRestore();
+  });
+
+  it("Warn calls console.warn", () => {
+    const spy = jest.spyOn(console, "warn").mockImplementation();
+    HFX.Logger.warn("Warn");
+    expect(console.warn).toHaveBeenCalledTimes(1);
+    spy.mockRestore();
+  });
+
+  it("Log calls console.log", () => {
+    const spy = jest.spyOn(console, "log").mockImplementation();
+    HFX.Logger.log("Log");
+    expect(console.log).toHaveBeenCalledTimes(1);
+    spy.mockRestore();
+  });
+
+  it("Debug doesn't call console.log", () => {
+    const spy = jest.spyOn(console, "log").mockImplementation();
+    HFX.Logger.debug("Debug", {a: "a"});
+    expect(console.log).toHaveBeenCalledTimes(0);
+    spy.mockRestore();
```

Comment:
This should be 1, not 0.

Reply:
It's dependent on the debugMode. Because of the toggle test and not cleaning up after that this be 0 or 1 depends on the tests ran and order.

Reply:
How do You recommend I set up this test correctly regarding the order at which they run?

Reply:
See the changes I pushed. This was only an issue because the toggleDebug test was toggling the debugMode variable an odd number of times (1). That meant a test of the debug method would have mixed results.